test/unit/runner/providers/ProviderRegistryTest.php

```diff
@@ -0,0 +1,135 @@
+            'module'      => 'taoQtiTest/runner/provider/qti',
+            'bundle'      => 'taoQtiTest/loader/qtiTestRunner.min',
+            'description' => 'QTI implementation of the test runner',
+            'category'    => 'runner',
+            'active'      => true,
+            'tags'        => [ 'core', 'qti', 'runner' ]
+        ], [
+            'id'          => 'qtiprint',
+            'name'        => 'QTI runner for paper',
+            'module'      => 'taoQtiPrint/runner/provider/qtiprint',
+            'bundle'      => 'taoQtiPrint/loader/qtiTestRunner.min',
+            'description' => 'QTI implementation of the test runner on paper',
+            'category'    => 'runner',
+            'active'      => true,
+            'tags'        => [ 'core', 'qti', 'runner', 'print' ]
+        ], [
+            'id'       => 'request',
+            'name'     => 'request communicator',
+            'module'   => 'core/communicator/request',
+            'bundle'   => 'loader/vendor.min',
+            'category' => 'communicator',
+            'active'   => true,
+            'tags'     => [ ]
+        ], [
+            'id'       => 'sockets',
+            'name'     => 'web sockets communicator',
+            'module'   => 'core/communicator/ws',
+            'bundle'   => 'loader/vendor.min',
+            'category' => 'communicator',
+            'active'   => true,
+            'tags'     => [ ]
+        ], [
+            'id'       => 'poll',
+            'name'     => 'poll communicator',
+            'module'   => "core/communicator/poll",
+            'bundle'   => 'loader/vendor.min',
+            'category' => 'communicator',
+            'active'   => true,
+            'tags'     => [ ]
+        ], [
+            'id' => 'qtiServiceProxy',
+            'module' => 'taoQtiTest/runner/proxy/qtiServiceProxy',
+            'bundle' => 'taoQtiTest/loader/qtiTestRunner.min',
+            'category' => 'proxy',
+            'active'   => true,
+            'tags'     => [ ]
+        ]
+    ];
+
+    /**
+     * Data provider
+     * @return array the data
+     */
+    public function categoryFindingProvider()
+    {
+        return [
+            [ null, 0 ],
+            [ '', 0 ],
+            [ 'foo', 0 ],
+            [ 'runner', 2 ],
+            [ 'communicator', 3 ],
+            [ 'proxy', 1 ],
+        ];
+    }
+
+    /**
+     * Test getting providers by category
+     * @dataProvider categoryFindingProvider
+     */
+    public function testGetByCategory($category, $expectedProviderCount)
+    {
+        $registry = $this->getMockBuilder(ProviderRegistry::class)
+                ->setMethods(['getMap', '__construct'])
+                ->disableOriginalConstructor()
+                ->getMock();
+        $registry->method('getMap')->willReturn(self::$map);
+
+        $providers = $registry->getByCategory($category);
+        $this->assertEquals(count($providers), $expectedProviderCount);
```

Comment:
Isn't it required to test that the correct providers get returned?

Reply:
you can use assertCount maybe ?

Reply:
I was hesitating  so yes